mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp

```diff
@@ -50,45 +50,54 @@ static unsigned short float_to_bfloat16(float src_val) {
   return target_val.ushortvec[1];
 }
 
-static unsigned short int_to_fp16(short src_val) {
-  unsigned short sign = src_val < 0;
-  unsigned short absx = ((unsigned short)src_val ^ -sign) + sign; // safe abs(x)
-  unsigned short tmp = absx, manbits = 0;
-  int exp = 0, truncated = 0;
-
-  // calculate the number of bits needed for the mantissa
-  while (tmp) {
-    tmp >>= 1;
-    manbits++;
-  }
-
-  // half-precision floats have 11 bits in the mantissa.
-  // truncate the excess or insert the lacking 0s until there are 11.
-  if (manbits) {
-    exp = 10; // exp bias because 1.0 is at bit position 10
-    while (manbits > 11) {
-      truncated |= absx & 1;
-      absx >>= 1;
-      manbits--;
-      exp++;
-    }
-    while (manbits < 11) {
-      absx <<= 1;
-      manbits++;
-      exp--;
+// Tables for float-to-fp16 conversion
```

Comment:
@yiqian1 could you document the source reference of how the tables are populated.